### DISCUSSION TITLE
Importing a .csv for data source

### DISCUSSION THREAD

**Discussion by tgerardot** [BigJk/snd#41] [Q&A]
Hey there! I've got a large list of spells I would love to import as a data source. I compiled it into a json format, then was able to create a csv file following the example. I believe the issue is that the number of columns in the csv far exceeds the 4 in the example? I'm having a difficult time importing it as a data source and I receive the error "header for values not present (record on line 3: wrong number of fields)".

Has anyone had this issue and been able to remedy it?

**Comment by BigJk** [comment]
Regarding the CSV problem I will investigate. Feel free to post a snippet of it here (first 10 lines) or send it to me via Discord 😄 

You can also just build your Data Source from the JSON. A data source is essentially 2 files (example: https://github.com/BigJk/snd-open5e/tree/main/BigJk_Open5e%20Weapons)

## meta.json

Just the basic infos

```
{
  "name": "Open5e Weapons",
  "slug": "open5e-weapons",
  "author": "BigJk",
  "description": "Imported weapon data from Open5e"
}
```

## entries.json

You entries as a array. Objects need a unique ``id``, some ``name`` and then the ``data`` object with whatever data it should hold. Fairly plain.

```json5
[
   {
      "id":"open5e/club",
      "name":"Club",
      "data": {
          // YOUR DATA HERE
      }
   },
   {
      "id":"open5e/dagger",
      "name":"Dagger",
      "data":{
          // YOUR DATA HERE
      }
   }
]
```

Then you can just import the folder 

**EDIT:** found the CSV bug and the fix is in the latest release.

**Reply by tgerardot** [reply]
Oh, yea that's definitely easy. I'll go this json route, thank you!

Happy the bu sounded like it was an easy fix.
